addon/object/index.js

```diff
@@ -55,8 +55,10 @@ export const action = decorator(function(target, key, desc) {
   // We must collapse the superclass prototype to make sure that the `actions`
   // object will exist. Since collapsing doesn't generally happen until a class is
   // instantiated, we have to do it manually.
-  if (typeof target.constructor.proto === 'function') {
-    target.constructor.proto();
+  let superClass = Object.getPrototypeOf(target.constructor);
+
+  if (superClass.hasOwnProperty('proto') && typeof superClass.proto === 'function') {
```

Comment:
I think we might want an `in` check? What if we have:

```
let Foo = Ember.Object.extend();
let Bar = class extends Foo {
  @action
  derp() {}
 };
let Baz = class extends Bar {
  @action
  quiz() {}
}
```

Baz should have two actions (derp and qux).